scripts/bravo_csv.py

```diff
@@ -303,29 +307,29 @@ def dilution(currentStep):
+                                art_tuple[1]['uri'].udf['Final Volume (uL)'] = min_vol_for_dilution
+                                logContext.write("WARN : Sample {0} located {1} {2} has a LOWER conc than {3}. Take {4} ul directly into the dilution plate.\n".format(art_tuple[1]['uri'].samples[0].name,art_tuple[0]['uri'].location[0].name, art_tuple[0]['uri'].location[1],min_required_conc,min_vol_for_dilution))
                             else:
                                 art_tuple[1]['uri'].udf['Volume to take (uL)'] = 0
                                 art_tuple[1]['uri'].udf['Final Concentration'] = 0
                                 art_tuple[1]['uri'].udf['Final Volume (uL)'] = 0
-                                logContext.write("ERROR : Sample {0} located {1} {2} has a LOWER conc than {3} and total volume less than {4} ul. It is skipped in dilution.\n".format(art_tuple[1]['uri'].samples[0].name,art_tuple[0]['uri'].location[0].name, art_tuple[0]['uri'].location[1],preset[0],preset[2]))
+                                logContext.write("ERROR : Sample {0} located {1} {2} has a LOWER conc than {3} and total volume less than {4} ul. It is skipped in dilution.\n".format(art_tuple[1]['uri'].samples[0].name,art_tuple[0]['uri'].location[0].name, art_tuple[0]['uri'].location[1],min_required_conc,min_vol_for_dilution))
                         # Case that sample concentration higher than the maximum conc for dilution
-                        elif art_tuple[1]['uri'].udf['Concentration'] > preset[1]:
+                        elif art_tuple[1]['uri'].udf['Concentration'] > max_conc_for_dilution:
                             art_tuple[1]['uri'].udf['Volume to take (uL)'] = 0
                             art_tuple[1]['uri'].udf['Final Concentration'] = 0
                             art_tuple[1]['uri'].udf['Final Volume (uL)'] = 0
-                            logContext.write("ERROR : Sample {0} located {1} {2} has a HIGHER conc than {3}. It is skipped in dilution.\n".format(art_tuple[1]['uri'].samples[0].name,art_tuple[0]['uri'].location[0].name, art_tuple[0]['uri'].location[1],preset[1]))
+                            logContext.write("ERROR : Sample {0} located {1} {2} has a HIGHER conc than {3}. It is skipped in dilution.\n".format(art_tuple[1]['uri'].samples[0].name,art_tuple[0]['uri'].location[0].name, art_tuple[0]['uri'].location[1],max_conc_for_dilution))
                         # Case that dilution will be done with 2uL sample
-                        elif art_tuple[1]['uri'].udf['Concentration'] <= preset[1] and art_tuple[1]['uri'].udf['Concentration'] > float(preset[0]*preset[2]/MIN_WARNING_VOLUME):
+                        elif art_tuple[1]['uri'].udf['Concentration'] <= max_conc_for_dilution and art_tuple[1]['uri'].udf['Concentration'] > float(min_required_conc*min_vol_for_dilution/MIN_WARNING_VOLUME):
```

Comment:
Why use `float()` like this here?

Reply:
This is because in the presets some values may be `int` when someone else put values there.

Reply:
Then shouldn't you convert the divisor or dividend to float and not the result?

```
float(min_required_conc*min_vol_for_dilution)/MIN_WARNING_VOLUME
```

Reply:
Yes, I will change it.

Reply:
Here's what I mean with Python 2:

```python
>>> 2*3/7
0
>>> float(2*3/7)
0.0
>>> float(2*3)/7
0.8571428571428571
```

Apparently you could also do:

```python
from __future__ import division
```

Reply:
That's a good solution! Thanks!